core/lib/tasks/migrations/migrate_address_names.rake

```diff
@@ -37,115 +38,115 @@ namespace :solidus do
+          # batch took 11 seconds
+          # batch took 12 seconds
+          # batch took 13 seconds
+          # batch took 12 seconds
+          # batch took 12 seconds
+          # batch took 14 seconds
+          # batch took 19 seconds
+          # batch took 20 seconds
+          # batch took 21 seconds
+          #
+          # Please note that the migration will be much faster when there's no index
+          # on the `name` column. For example, with mysql each batch takes exactly
+          # the same time:
+          #
+          # Updating 1000000 records in batches of 200000, this may take a while...
+          # batch took 17 seconds
+          # batch took 17 seconds
+          # batch took 17 seconds
+          # batch took 16 seconds
+          # batch took 17 seconds
+          #
+          # So, if special need arise, one can drop the index added with migration
+          # 20210122110141_add_name_to_spree_addresses.rb and add the index later,
+          # in non blocking ways. For postgresql:
+          # add_index(:spree_addresses, :name, algorithm: :concurrently)
+          #
+          # For mysql 5.6:
+          # sql = "ALTER TABLE spree_addresses ADD INDEX index_spree_addresses_on_name (name), ALGORITHM=INPLACE, LOCK=NONE;"
+          # ActiveRecord::Base.connection.execute sql
+          #
+          puts '  Data migration will happen in batches. The default value is 100_000, which should take less than 20 seconds on mysql or postgresql.'
+          size = shell.ask('  Please enter a different batch size, or press return to confirm the default: ')
+          size = (size.presence || 100_000).to_i
+
+          abort "  Invalid batch size number #{size}, please run the task again." unless size.positive?
 
-            puts "We're going to migrate #{count} records in batches of #{size}"
-            answer = shell.ask('do you want to proceed?', limited_to: ['Y', 'N'], case_insensitive: true)
-            if answer == 'Y'
-              puts "Updating #{count} records in batches of #{size}, this may take a while..."
+          puts "  We're going to migrate #{count} records in batches of #{size}."
+          answer = shell.ask('  Do you want to proceed?', limited_to: ['Y', 'N'], case_insensitive: true)
+          if answer == 'Y'
+            puts "  Updating #{count} records in batches of #{size}, this may take a while..."
 
-              Spree::AddressForMigration.unscoped.where(name: [nil, '']).in_batches(of: size).each do |batch|
-                now = Time.zone.now
-                batch.update_all(concat_statement)
-                puts "batch took #{(Time.zone.now - now).to_i} seconds"
-              end
+            Spree::AddressForMigration.unscoped.where(name: [nil, '']).in_batches(of: size).each_with_index do |batch, index|
+              now = Time.zone.now
+              batch.update_all(concat_statement)
+              puts "  Batch ##{index + 1} done in #{(Time.zone.now - now).to_i} seconds."
```

Comment:
Would it be helpful to also include the total number of batches here (i.e., Batch 1/50)? I'm not sure if that's easy to do with ActiveRecord, but it would be a nice addition.